src/ruleset/v3/ruleset.ts

```diff
@@ -24,5 +26,19 @@ export const v3CoreRuleset = {
         function: operationMessagesUnambiguity,
       },
     },
+    'asyncapi3-required-operation-channel-unambiguity': {
+      description: 'The "channel" field of an operation under the root "operations" object must always reference a channel under the root "channels" object.',
+      severity: 'error',
+      recommended: true,
+      resolved: false, // We use the JSON pointer to match the channel.
+      given: '$.operations.*',      
+      then: {
+        field: 'channel.$ref',
+        function: pattern,
+        functionOptions: {
+          match: '#\\/channels\\/', // If doesn't match, rule fails.
```

Comment:
@fmvilas @jonaslagoni I simplified this by using this `pattern` function Spectral gives us. It allows to match by a regex applied to a field, in this case `channel.$ref`. 
In consequence, I removed the function I created in favor of this.

Reply:
Much better 👍